### DISCUSSION TITLE
support reaction (in the y direction)

### DISCUSSION THREAD

**Discussion by sinamitnn** [FEALiTE/FEALiTE2D#15] [General]
Hi, 
first of all i would like to thank you for this great work and for the sharing.

I have a question about the reaction forces: 
Even though i do not have any load in the y-direction (and/or any moment), i get -very small- reaction forces in the y-direction.
They should be actually 0. 
Did you also exprience such results? 

Thanks.

-Sinan

**Comment by SaMohamed** [comment]
Hi Sinan,
Thank you for reaching me.
Can you share your code with me? so I can investigate it?

**Reply by sinamitnn** [reply]
Hi Mohammed,
i thank you for your fast reply.

Below you can find an example where you can observe the problem:

********

            FEALiTE2D.Structure.Structure structure = new FEALiTE2D.Structure.Structure();

            Node2D n1 = new Node2D(0d, 0d, "n1");
            Node2D n2 = new Node2D(0d, 0.12, "n2");
            Node2D n3 = new Node2D(0d, 1.12, "n3");
            Node2D n4 = new Node2D(0d, 2.12, "n4");
            Node2D n5 = new Node2D(0d, 2.6, "n5");

            Node2D n6 = new Node2D(1.1, 0d, "n6");
            Node2D n7 = new Node2D(1.1, 0.12, "n7");
            Node2D n8 = new Node2D(1.1, 0.62, "n8");
            Node2D n9 = new Node2D(1.1, 1.62, "n9");
            Node2D n10 = new Node2D(1.1, 2.12, "n10");
            Node2D n11 = new Node2D(1.1, 2.6, "n11");




            n1.Support = new NodalSupport(true, true, false);
            n6.Support = new NodalSupport(true, true, false);

            structure.AddNode(n1, n2, n3, n4, n5, n6, n7, n8, n9, n10, n11);


            IMaterial material = new GenericIsotropicMaterial() { E = 210000000, U = 0.3, Label = "Steel", Alpha = 1, Gama = 1, MaterialType = MaterialType.Steel };



         


            IFrame2DSection section = new Generic2DSection(0.000248, 1, 1, 0.0000003615, 1, 1, 1, 1, material);
            IFrame2DSection diagonalsection = new Generic2DSection(0.000248, 1, 1, 0.000000361, 1, 1, 1, 1, material);


            //MessageBox.Show(section.A.ToString());
            //MessageBox.Show(section.Ix.ToString());


            //Column
            FrameElement2D e1 = new FrameElement2D(n1, n2, "e1") { CrossSection = section };
            FrameElement2D e2 = new FrameElement2D(n2, n3, "e2") { CrossSection = section };
            FrameElement2D e3 = new FrameElement2D(n3, n4, "e3") { CrossSection = section };
            FrameElement2D e4 = new FrameElement2D(n4, n5, "e4") { CrossSection = section };

            FrameElement2D e5 = new FrameElement2D(n6, n7, "e5") { CrossSection = section };
            FrameElement2D e6 = new FrameElement2D(n7, n8, "e6") { CrossSection = section };
            FrameElement2D e7 = new FrameElement2D(n8, n9, "e7") { CrossSection = section };
            FrameElement2D e8 = new FrameElement2D(n9, n10, "e8") { CrossSection = section };
            FrameElement2D e9 = new FrameElement2D(n10, n11, "e9") { CrossSection = section };

            //Diagonal
            FrameElement2D e10 = new FrameElement2D(n7, n2, "e10") { CrossSection = diagonalsection, EndRelease = Frame2DEndRelease.FullRelease };
            FrameElement2D e11 = new FrameElement2D(n2, n8, "e11") { CrossSection = diagonalsection, EndRelease = Frame2DEndRelease.FullRelease };
            FrameElement2D e12 = new FrameElement2D(n8, n3, "e12") { CrossSection = diagonalsection, EndRelease = Frame2DEndRelease.FullRelease };
            FrameElement2D e13 = new FrameElement2D(n3, n9, "e13") { CrossSection = diagonalsection, EndRelease = Frame2DEndRelease.FullRelease };
            FrameElement2D e14 = new FrameElement2D(n9, n4, "e14") { CrossSection = diagonalsection, EndRelease = Frame2DEndRelease.FullRelease };
            FrameElement2D e15 = new FrameElement2D(n4, n10, "e15") { CrossSection = diagonalsection, EndRelease = Frame2DEndRelease.FullRelease };

            //
            structure.AddElement(new[] { e1, e2, e3, e4, e5, e6, e7, e8, e9, e10, e11, e12, e13, e14, e15 });

            LoadCase loadCase = new LoadCase("live", LoadCaseType.Live);
            structure.LoadCasesToRun.Add(loadCase);



            e2.Loads.Add(new FramePointLoad(0d, -3, 0d, e2.Length  /2, LoadDirection.Global, loadCase));
            e2.Loads.Add(new FramePointLoad(0d, -3, 0d, e2.Length  /2, LoadDirection.Global, loadCase));
            e2.Loads.Add(new FramePointLoad(0d, -3, 0d, e2.Length / 2, LoadDirection.Global, loadCase));
           


            e3.Loads.Add(new FramePointLoad(0d, -3, 0d, e3.Length / 2, LoadDirection.Global, loadCase));
            e3.Loads.Add(new FramePointLoad(0d, -3, 0d, e3.Length / 2, LoadDirection.Global, loadCase));
            e3.Loads.Add(new FramePointLoad(0d, -3, 0d, e3.Length / 2, LoadDirection.Global, loadCase));




            //fem model loads for the second column

            e6.Loads.Add(new FramePointLoad(0d, -3, 0d, e6.Length / 2, LoadDirection.Global, loadCase));
            e6.Loads.Add(new FramePointLoad(0d, -3, 0d, e6.Length / 2, LoadDirection.Global, loadCase));
            e6.Loads.Add(new FramePointLoad(0d, -3, 0d, e6.Length / 2, LoadDirection.Global, loadCase));



            e7.Loads.Add(new FramePointLoad(0d, -3, 0d, e7.Length / 2, LoadDirection.Global, loadCase));
            e7.Loads.Add(new FramePointLoad(0d, -3, 0d, e7.Length / 2, LoadDirection.Global, loadCase));
            e7.Loads.Add(new FramePointLoad(0d, -3, 0d, e7.Length / 2, LoadDirection.Global, loadCase));


            e8.Loads.Add(new FramePointLoad(0d, -3, 0d, e8.Length / 2, LoadDirection.Global, loadCase));
            e8.Loads.Add(new FramePointLoad(0d, -3, 0d, e8.Length / 2, LoadDirection.Global, loadCase));
            e8.Loads.Add(new FramePointLoad(0d, -3, 0d, e8.Length / 2, LoadDirection.Global, loadCase));




            structure.LinearMesher.NumberSegements = 10;
            structure.Solve();

            MessageBox.Show(structure.Results.GetSupportReaction(n1, loadCase).ToString());
            MessageBox.Show(structure.Results.GetSupportReaction(n6, loadCase).ToString());


*****

Thanks again. 

BR, 

Sinan.

**Reply by SaMohamed** [reply]
Dear Sinan,

In this example that you gave me, I can see that you put a load in y direction on element e2, e3, e6, e7 and e8, and the load is in global coordinate system, so you should get reactions in y direction.
and i want to note that you tripled the load on all of these elements

**Reply by sinamitnn** [reply]
Hello Mohammed, 

actually i had to it for the x-direction.
Even though i do not have any load in the x-direction, i get support reactions in the x-direction. 

![grafik](https://github.com/FEALiTE/FEALiTE2D/assets/147400466/d28e7595-f397-4c82-b1b3-33e206e8a885)

Thanks again. 

Sinan.

**Reply by SaMohamed** [reply]
is this the model you are trying to achieve?
![image](https://github.com/FEALiTE/FEALiTE2D/assets/21183259/22f4468b-c299-421b-8f0b-9f6af950a34a)

**Reply by sinamitnn** [reply]
no. What i want to model, you can find below. 
![grafik](https://github.com/FEALiTE/FEALiTE2D/assets/147400466/bff5e1f0-c20f-40f8-9b7e-cedd419fbd5b)

The number of the point loads for this example is not important. If i have only one point load, i get also reaction in the x direction. 
![grafik](https://github.com/FEALiTE/FEALiTE2D/assets/147400466/e6c8d976-370d-4622-815c-a165b4d7dd3b)

**Reply by SaMohamed** [reply]
Hi Sinan,

I see this is normal because you have diagonals and they work as strut and tie, so you'll get a reaction in x direction at both supports and they'll be in opposite direction.

this image is from SAP2000 and it show and it confirms what I say
![image](https://github.com/FEALiTE/FEALiTE2D/assets/21183259/1f208014-21b3-481f-913f-998a3f45f434)

**Reply by sinamitnn** [reply]
Hello Salah, 
thanks again for the fast reply and for the explanation.
According to me understanding the loads should flow directly to the supports.
i am also going to make a check-calculation in rstab. 
May be i should open my structural analysis book from the bachelor times:) 
Thanks again for sharing this great code.

-Sinan

**Reply by SaMohamed** [reply]
it's my pleasure and I'm waiting for your update after checking from your side.

**Comment by sinamitnn** [comment]
Hello Salah, 

finally i made a check-calculation with R-Stab.

The/your code tells actually the truth:) 
![grafik](https://github.com/FEALiTE/FEALiTE2D/assets/147400466/14f97e3c-0d29-477d-9a76-2a9af50e2ce5)

Thanks again.

BR, 

Sinan.

**Reply by SaMohamed** [reply]
Hi Sinan,
Thank you for your time and to make verification
